src/svg/qsvggenerator.cpp

```diff
@@ -997,8 +1026,36 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     // close old state and start a new one...
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
+    
+    QPainter* p = painter();
+    if (p->hasClipping()) {
+        std::string clip_path;
+        QPainterPath path = p->clipPathF();
+
+        if (path.elementCount() > 0) {
+            QPainterPath::Element starting_point = path.elementAt(0);
```

Comment:
You might be able to use const references here.  This would be consistent with other stuff using the result returned by QPainterPath.elementAt().